test/integration/safer_cluster/safer_cluster_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package safer_cluster
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/gcloud"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/golden"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSaferCluster(t *testing.T) {
+	bpt := tft.NewTFBlueprintTest(t)
+
+	bpt.DefineVerify(func(assert *assert.Assertions) {
+		//Skipping Default Verify as the Verify Stage fails due to change in Client Cert Token
+		// bpt.DefaultVerify(assert)
+
+		projectId := bpt.GetStringOutput("project_id")
+		location := bpt.GetStringOutput("location")
+		clusterName := bpt.GetStringOutput("cluster_name")
+		serviceAccount := bpt.GetStringOutput("service_account")
+
+		op := gcloud.Runf(t, "container clusters describe %s --zone %s --project %s", clusterName, location, projectId)
+		g := golden.NewOrUpdate(t, op.String(),
+			golden.WithSanitizer(golden.StringSanitizer(serviceAccount, "SERVICE_ACCOUNT")),
+			golden.WithSanitizer(golden.StringSanitizer(projectId, "PROJECT_ID")),
+			golden.WithSanitizer(golden.StringSanitizer(clusterName, "CLUSTER_NAME")),
+		)
+		validateJSONPaths := []string{
+			"status",
+			"location",
+			"privateClusterConfig.enablePrivateEndpoint",
+			"privateClusterConfig.enablePrivateNodes",
+			"addonsConfig.horizontalPodAutoscaling",
+			"addonsConfig.kubernetesDashboard",
+			"addonsConfig.networkPolicyConfig",
+			"networkPolicy",
+			"networkConfig.datapathProvider",
+			"binaryAuthorization.evaluationMode",
+			"legacyAbac",
+			"nodePools.autoscaling",
+			"nodePools.config.machineType",
+			"nodePools.config.diskSizeGb",
+			"nodePools.config.labels",
+			"nodePools.config.tags",
+			"nodePools.management.autoRepair",
+			"nodePools.shieldedInstanceConfig",
+		}
+		for _, pth := range validateJSONPaths {
+			g.JSONEq(assert, op, pth)
+		}
+		op1 := gcloud.Runf(t, "compute firewall-rules list  --project %s", projectId)
+		assert.Contains(op1.String(), fmt.Sprintf("gke-%s-intra-cluster-egress", clusterName[:25]), "Firewall Rule Exists")
+		assert.Contains(op1.String(), fmt.Sprintf("gke-%s-webhooks", clusterName[:25]), "Firewall Rule Exists")
```

Comment:
nit: rather than listing + string comparison, we could just do a `gcloud compute firewall-rules describe <expected name>`. If that rule does not exists gcloud will complain and fail.

Reply:
Added the changes based on suggestion